### DISCUSSION TITLE
Does the JavaScript expression support allow presets to be saved as preferences?

### DISCUSSION THREAD

**Discussion by SignpostMarv** [mifi/lossless-cut#2035] [Q&A] [answered]
I prefer not to update LosslessCut until absolutely necessary, but the JavaScript expression support has me intrigued; Due to LC not currently (on 3.59.1.0) supporting persistence of segment selection I typically prefix segments I don't want to export with `__dne__` (do not export) so if I shut down LC (or the system reboots while it's sleeping) it's relatively easy for me to catch segments I don't want to use.

If I can save javascript expressions as presets then I'll be jumping on that, otherwise I'll be waiting a bit before updating.

**Comment by mifi** [comment, marked as answer]
> Due to LC not currently (on 3.59.1.0) supporting persistence of segment selection

please leave a thumbs up in #1329

> If I can save javascript expressions as presets then I'll be jumping on that, otherwise I'll be waiting a bit before updating.

That's a great idea. I've added it to #371

**Comment by devtweakr** [comment]
when select by expression dialog is open:

1. remember last used
2. add a list of presets (list could be same style as segments in main UI with + and - buttons as well)
3. add an option to "select" or "deselect" (presets also save this option)
